### DISCUSSION TITLE
Entity linker queries of aliases and entity descriptions

### DISCUSSION THREAD

**Discussion by sednaasil** [explosion/spaCy#6780] [Help: Coding & Implementations] [answered]
Hi:  

I have established a working entity linking tool using WikiData/Wikipedia.  I am interested in added the known aliases and entity descriptions to the structured output I am generating.  However, I can't seem to find a function to query an entity ID to get description or a list of aliases.  Does this function exist?

Thanks so much, in advance.

Kindest regards,
L

**Comment by svlandeg** [comment, marked as answer]
Hi!

Because the Knowledge Base can potentially grow quite big, we've opted to make it as efficient as possible for the retrieval task required by the Entity Linker - which is retrieving possible KB ID's, given a certain alias/mention. Unfortunately this also means that querying the other way around (ID -> set of aliases) will be inefficient in the current implementation. To add insult to injury, the description is never stored in its original string form, only the vectorized form of it. Again, because of efficiency reasons in the core implementation.

One solution for your use-case could be to create a subclass of `KnowledgeBase` that reimplements the `add_entity`, `set_entities`, `add_alias` and `append_alias` methods, stores the additional information you want in whichever way you want it, and then calls the corresponding methods of the base class.

If you used the example Wikipedia/Wikidata scripts that we provided (https://github.com/explosion/projects/tree/master/nel-wikipedia), this script actually stores some intermediate processing files to disk, which contain the information you need. So you could also query your information directly from these files.

The title/name of an entity:
```
cat entity_defs.csv

WP_title|WD_id
Belgium|Q31
Happiness|Q8
George Washington|Q23
```

The description of an entity:
```
cat entity_descriptions.csv

WD_id|description
Q31|federal constitutional monarchy in Western Europe
Q8|mental or emotional state of well-being characterized by pleasant emotions
Q23|First President of the United States
```
And the aliases of a given entity:
```
cat entity_alias.csv

WD_id|alias
Q31|Kingdom of Belgium
Q31|be
Q31|🇧🇪

Q8|😄
Q8|:)
Q8|😃
Q8|joy
Q8|happy

Q23|Washington
Q23|President Washington
Q23|G. Washington
Q23|Father of the United States
````

Hope that works for you!
